src/store.cc

```diff
@@ -1442,8 +1442,11 @@ StoreEntry::updateOnNotModified(const StoreEntry &e304)
     // update reply before calling timestampsSet() below
     const auto &oldReply = mem_obj->freshestReply();
     const auto updatedReply = oldReply.recreateOnNotModified(e304.mem_obj->baseReply());
-    if (updatedReply) // HTTP 304 brought in new information
+    if (updatedReply) { // HTTP 304 brought in new information
+        if (updatedReply->prefixLen() > Config.maxReplyHeaderSize)
+            throw TextException("cannot update the cached response because its header would become too large", Here());
```

Comment:
Just polishing, with more details:

```suggestion
            throw TextException(ToSBuf("cannot update the cached response because its updated ",
                                                          updatedReply->prefixLen(), "-byte header would exceed ",
                                                          Config.maxReplyHeaderSize, "-byte reply_header_max_size"), Here());
```

The above suggestion may need minor adjustments and formatting.